### PR TITLE
HS-30: UI with keycloak

### PR DESCRIPTION
**Pull request by mikewrosey**
## JIRA Link
https://issues.opennms.org/browse/HS-30

## Checklist
* [x] JIRA has been updated to 'in review'.
* [X] Unit tests have been added as part of the PR.
* [X] Manual tests have been completed, and notes added to JIRA.
* [x] Details have been provided to docs team (if necessary)
* [X] Screenshot(s) included (applicable to changes with a visual impact).

## Details of the change
 - Adds keycloak integration endpoints.
 - Uses custom login screen to retrieve token.
 
screenshot-localhost_3001-2022 05 06-15_27_24

### CODE REVIEW

File: ui/src/containers/Login.vue [hunk 1]
```diff
@@ -34,14 +40,20 @@
...
+import { useAuthStore } from '@/store/authStore'
+const authStore = useAuthStore()
 
+const usernameInput = ref()
 const username = ref('')
 const password = ref('')
 const usernameError = ref()
 const passwordError = ref()
 
+onKeyStroke('Enter', (e) => {
```
Comment (geraldhumphries) [inline ui/src/containers/Login.vue:52]: What about using `onLoginBtnClick` in the `<form>` tag of the template, similar to the button?

File: ui/src/containers/Login.vue [hunk 2]
```diff
@@ -59,9 +71,11 @@ const onLoginBtnClick = () => {
 
   // submit if form is valid
   if (!usernameError.value && !passwordError.value) {
-    loginStore.login(username.value, password.value)
+    authStore.login(username.value, password.value)
   }
 }
+
+onMounted(() => usernameInput.value.focus())
```
Comment (geraldhumphries) [inline ui/src/containers/Login.vue:78]: Can you use the `autofocus` attribute on the input, or is that not possible with FeatherInput?

Comment (mikewrosey) [reply ui/src/containers/Login.vue]: Yup, works with feather

File: ui/src/containers/Login.vue (hunk 1 above)
Comment (mikewrosey) [reply ui/src/containers/Login.vue]: done

**Review by geraldhumphries — APPROVED** [review]

**Pull request merged**